src/Processors/QueryPlan/AggregatingStep.cpp

```diff
@@ -770,6 +767,18 @@ void AggregatingStep::serialize(Serialization & ctx) const
     for (const auto & key : params.keys)
         writeStringBinary(key, ctx.out);
 
+    if (!grouping_sets_params.empty())
+    {
+        writeVarUInt(grouping_sets_params.size(), ctx.out);
```

Comment:
I think this should be moved outside the `if` scope and always written even if it is 0, so that it is not does not collide with next field during deserialization.

Reply:
Ok, this is addressed by a flag